File: src/main/java/org/usfirst/frc/team1458/lib/core/BaseAutoMode.java
package org.usfirst.frc.team1458.lib.core;

/**
 * Base class for autonomous mode. This is an abstract
 * implementation of AutoMode with some utility functions, but it must be
 * extended to be used.
 *
 * @author asinghani
 */
public abstract class BaseAutoMode implements AutoMode {

	@Override
	public String getName() {
		return "\"" + getClass().getSimpleName() + "\"";
	}

	@Override
	public String toString() {
		return getName();
	}

	/**
	 * The {@link #auto()} function must be implemented by the class extending
	 * SampleAutoMode
	 */
	@Override
	public abstract void auto();
}

File: src/main/java/org/usfirst/frc/team1458/lib/core/BaseAutoMode.kt
package org.usfirst.frc.team1458.lib.core

/**
 * Base class for autonomous mode. This is an abstract
 * implementation of AutoMode with some utility functions, but it must be
 * extended to be used.
 *
 * @author asinghani
 */
abstract class BaseAutoMode : AutoMode {

    override val name: String
        get() = "\"" + javaClass.simpleName + "\""

    override fun toString(): String {
        return name
    }

    /**
     * The [.auto] function must be implemented by the class extending
     * SampleAutoMode
     */
    abstract override fun auto()
}